packages/vm/src/cache.rs

```diff
@@ -116,6 +119,41 @@ where
         }
     }
 
+    /// Pins a Module that was previously stored via save_wasm.
+    ///
+    /// The module is lookup first in the memory cache, and then in the file system cache.
+    /// If not found, the code is loaded from the file system, compiled, and stored into the
+    /// pinned cache.
+    /// If the given ID is not found, or the content does not match the hash (=ID), an error is returned.
+    pub fn pin(&mut self, checksum: &Checksum) -> VmResult<()> {
+        // Try to get module from the memory cache
+        if let Some(module) = self.memory_cache.load(checksum)? {
+            self.stats.hits_memory_cache += 1;
+            return self.pinned_memory_cache.store(checksum, module);
+        }
+
+        // Try to get module from file system cache
+        let store = make_runtime_store(self.instance_memory_limit);
```

Comment:
`self.instance_memory_limit` should never be `None`. Do we still need the `Option` in `make_runtime_store`? Or can we revert to the original signature?

Reply:
You're right that with the current use cases there's no need to support an "unlimited" cache.

I still think that it could be useful, but, given that's not needed, I'm OK with reverting to `Size` here.

Reply:
If you like we can keep the API of `make_runtime_store` with an `Option` for consistency. Then we can easily check we always pass `make_runtime_store(Some(limit))` in the caller code.

Reply:
Yes, that would be better. In fact `make_runtime_store()` is being used in the store tests, and it's clearer to pass `None` than `Some(0)` when creating a no limit store.